src/python_framework_v01/nhd_network_utilities_v01.py

```diff
@@ -638,3 +669,128 @@ def set_networks(supernetwork="", geo_input_folder=None, verbose=True, debugleve
         supernetwork_data=supernetwork_data, verbose=verbose, debuglevel=debuglevel
     )
     return supernetwork_data, supernetwork_values
+
+
+def read_waterbody_df(parm_file, lake_id_mask=None):
+    ds = xr.open_dataset(parm_file)
+    df1 = ds.to_dataframe().set_index("lake_id")
+    if lake_id_mask:
+        df1 = df1.loc[lake_id_mask, :]
+    return df1
+
+
+def get_ql_from_wrf_hydro(ql_files):
+    li = []
+
+    for filename in ql_files:
+        ds = xr.open_dataset(filename)
+        df1 = ds.to_dataframe()
+
+        li.append(df1)
+
+    frame = pd.concat(li, axis=0, ignore_index=False)
+    mod = frame.reset_index()
+    ql = mod.pivot(index="station_id", columns="time", values="q_lateral")
+
+    return ql
+
+
+def get_stream_restart_from_wrf_hydro(
+    channel_initial_states_file,
+    crosswalk_file,
+    channel_ID_column,
+    us_flow_column="qlink1",
+    ds_flow_column="qlink2",
+    depth_column="hlink",
+    default_us_flow_column="qu0",
+    default_ds_flow_column="qd0",
+    default_depth_column="h0",
+):
+    """
+    channel_initial_states_file: WRF-HYDRO standard restart file
+    us_flow_column: column in the restart file to use for upstream flow initial state
+    ds_flow_column: column in the restart file to use for downstream flow initial state
+    depth_column: column in the restart file to use for depth initial state
+    crosswalk_file: File containing channel IDs IN THE ORDER of the Restart File
+    channel_ID_column: field in the crosswalk file to assign as the index of the restart values
+    """
+
+    xds = xr.open_dataset(crosswalk_file)
+
+    xdf = xds.to_dataframe()
+    xdf = xdf.reset_index()
+    xdf = xdf[[channel_ID_column]]
+
+    qds = xr.open_dataset(channel_initial_states_file)
+    qdf = qds.to_dataframe()
```

Comment:
We should get avoid this dataframe creation. We are essentially reading the entire file into memory here when we only only want to read a couple of variables.
```python
qdf = qds[[us_flow_column, ds_flow_column, depth_column]].to_dataframe()
```